src/main/java/org/mule/tools/apikit/output/MuleConfigGenerator.java

```diff
@@ -65,30 +74,55 @@ public MuleConfigGenerator(List<ApikitMainFlowContainer> apis, List<GenerationMo
   }
 
   public List<MuleConfig> generate() {
-    List<MuleConfig> configs = new ArrayList<>();
+    Set<MuleConfig> muleConfigs = new HashSet<>();
     if (flowEntries.isEmpty()) {
       apis.forEach(api -> {
-        MuleConfig muleConfig = api.getMuleConfig() != null ? api.getMuleConfig() : createMuleConfig(api);
-        updateApikitConfig(api, muleConfig);
-        configs.add(muleConfig);
+        MuleConfig mainMuleConfig = api.getMuleConfig() != null ? api.getMuleConfig() : createMuleConfig(api);
+        Optional<MuleConfig> apikitConfigMuleConfig = updateApikitConfig(api, mainMuleConfig);
+        muleConfigs.add(mainMuleConfig);
+        addApikitConfig(muleConfigs, apikitConfigMuleConfig);
       });
     } else {
-      Set<MuleConfig> muleConfigs = new HashSet<>();
-      ApikitMainFlowContainer api = flowEntries.get(0).getApi();
-      MuleConfig muleConfig = api.getMuleConfig() != null ? api.getMuleConfig() : createMuleConfig(api);
-
+      ApikitMainFlowContainer api = flowEntries.stream().findFirst().get().getApi();
+      MuleConfig mainMuleConfig = api.getMuleConfig() != null ? api.getMuleConfig() : createMuleConfig(api);
+      Optional<MuleConfig> apikitConfig = updateApikitConfig(api, mainMuleConfig);
       for (GenerationModel flowEntry : flowEntries) {
         Element apikitFlowScope = new APIKitFlowScope(flowEntry, isMuleEE()).generate();
-
-        int newFlowPositionIndex = getLastFlowIndex(muleConfig.getContentAsDocument()) + 1;
-        muleConfig.getContentAsDocument().getRootElement().getContent().add(newFlowPositionIndex, apikitFlowScope);
-        muleConfig.addFlow(new Flow(apikitFlowScope));
-        updateApikitConfig(api, muleConfig);
-        muleConfigs.add(muleConfig);
+        int newFlowPositionIndex = getLastFlowIndex(mainMuleConfig.getContentAsDocument()) + 1;
+        mainMuleConfig.getContentAsDocument().getRootElement().getContent().add(newFlowPositionIndex, apikitFlowScope);
+        mainMuleConfig.addFlow(new Flow(apikitFlowScope));
+        muleConfigs.add(mainMuleConfig);
+        addApikitConfig(muleConfigs, apikitConfig);
       }
-      configs.addAll(muleConfigs);
     }
-    return configs;
+    return new ArrayList<>(muleConfigs);
+  }
+
+  private void addApikitConfig(Set<MuleConfig> muleConfigs, Optional<MuleConfig> apikitConfigMuleConfig) {
+    if (apikitConfigMuleConfig.isPresent()) {
+      muleConfigs.add(apikitConfigMuleConfig.get());
+    }
+  }
+
+  private MuleConfig retrieveApikitConfigMuleConfig(MuleConfig mainMuleConfig) {
+    MuleConfig apikitConfigMuleConfig = mainMuleConfig;
```

Comment:
why are you using a new variable here ? the name apikitConfigMuleConfig is not clear